### ISSUE TITLE
ember client - Ended of Stream Looking for Response Prelude

### ISSUE THREAD

**Issue by tpolecat** [http4s/http4s#3667]
I'm getting this error pretty often … must be a client error since it's parsing a response .. I guess?

```
org.http4s.ember.core.Parser$Response$RespPrelude$RespPreludeError: Received Error while parsing prelude - Reached Ended of Stream Looking for Response Prelude
    at org.http4s.ember.core.Parser$Response$RespPrelude$.$anonfun$parsePrelude$6(Parser.scala:422)
    at org.http4s.ember.core.Parser$Response$RespPrelude$.$anonfun$parsePrelude$6$adapted(Parser.scala:401)
    at fs2.Pull$.$anonfun$flatMap$1(Pull.scala:85)
    at fs2.internal.FreeC$$anon$1.cont(Algebra.scala:54)
    at fs2.internal.FreeC$ViewL$$anon$9$$anon$10.<init>(Algebra.scala:243)
    at fs2.internal.FreeC$ViewL$$anon$9.cont(Algebra.scala:243)
```

**Comment by ChristopherDavenport** [comment]
This would be the server killing an idle connection and it ending up using the connection presently. 

We should be handling this gracefully and retrying.

**Comment by RaasAhsan** [comment]
This should now be fixed by #5064
